### DISCUSSION TITLE
api.runOnBackend error says don't use `async` but it's not used

### DISCUSSION THREAD

**Discussion by maphew** [TriliumNext/Trilium#6488] [Q&A] [answered]
### TriliumNext Version

0.90.7-beta

### What operating system are you using?

Windows

### What is your setup?

Local (no sync)

### Operating System Version

win 11 pro (10.0.22631 N/A Build 22631)

### Description

I'm attempting to get _[command-palette](https://github.com/justyns/trilium-scripts/tree/master/command-palette)_ script working with Trilium Next 0.90.7-beta and am getting this error:

> Error: You're passing an async function to api.runOnBackend() which will likely not work as you intended. Either make the function synchronous (by removing 'async' keyword), or use api.runAsyncOnBackendWithManualTransactionHandling()

Likely the error is their responsibility however in investigating I don't see how to follow the advice in the error message, so maybe the problem is shared.

The code triggering the message appears to be:
```js
async function getAvailableNotes(query) {
  const noteObjs = await api.runOnBackend(searchNotes, [query]);
  return noteObjs;
}
//...snip...
  async showCommands(query) {
    const commands = await api.runOnBackend(getAvailableCommands, [query.trim()]);
    await this.renderItems(commands);
  }
```

However `async` keyword is not being used, so how can it be removed?

---
_xref https://github.com/justyns/trilium-scripts/issues/12_

**Comment by maphew** [comment]
I'm betraying my javascript ignorance, and readng incomprehension: the functions begin with `async`! I was looking for a parameter. At any rate, my question remains: what's the proper way to deal with this?

**Reply by eliandoran** [reply, marked as answer]
The way to handle it is described in the error.

Instead of using `runOnBackend` which requires a non-async function, you can use `runAsyncOnBackendWithManualTransactionHandling` instead. This differentiation is made because you're not supposed to do any database operations while running asynchronously.

**Reply by maphew** [reply]
Thank you!

I got this to work by editing the source code and changing `runOnBackend` to `runAsyncOnBackendWithManualTransactionHandling`. 2 instances in each script.

It seemed to be necessary to close and restart Trilium for the change to be noticed, though maybe force reload frontend would have been enough (`Ctrl-Shif-R`).
